tls/t/test_mpi_math.c

```diff
@@ -560,6 +559,99 @@ mpi_elementary(void)
 	free(B);
 }
 
+/**
+ * Copy of ecp256_to_mont().
+ */
+static void
+__to_mont(unsigned long *__restrict r, const unsigned long *__restrict a)
+{
+	long t[8], a32[8], o;
+
+	a32[0] = a[0] & 0xffffffff;
+	a32[1] = a[0] >> 32;
+	a32[2] = a[1] & 0xffffffff;
+	a32[3] = a[1] >> 32;
+	a32[4] = a[2] & 0xffffffff;
+	a32[5] = a[2] >> 32;
+	a32[6] = a[3] & 0xffffffff;
+	a32[7] = a[3] >> 32;
+
+	/*  1  1  0 -1 -1 -1 -1  0 */
+	t[0] = 0 + a32[0] + a32[1] - a32[3] - a32[4] - a32[5] - a32[6];
+	/*  0  1  1  0 -1 -1 -1 -1 */
+	t[1] = 0 + a32[1] + a32[2] - a32[4] - a32[5] - a32[6] - a32[7];
+	/*  0  0  1  1  0 -1 -1 -1 */
+	t[2] = 0 + a32[2] + a32[3] - a32[5] - a32[6] - a32[7];
+	/* -1 -1  0  2  2  1  0 -1 */
+	t[3] = 0 - a32[0] - a32[1] + 2 * a32[3] + 2 * a32[4] + a32[5] - a32[7];
+	/*  0 -1 -1  0  2  2  1  0 */
+	t[4] = 0 - a32[1] - a32[2] + 2 * a32[4] + 2 * a32[5] + a32[6];
+	/*  0  0 -1 -1  0  2  2  1 */
+	t[5] = 0 - a32[2] - a32[3] + 2 * a32[5] + 2 * a32[6] + a32[7];
+	/* -1 -1  0  0  0  1  3  2 */
+	t[6] = 0 - a32[0] - a32[1] + a32[5] + 3 * a32[6] + 2 * a32[7];
+	/*  1  0 -1 -1 -1 -1  0  3 */
+	t[7] = 0 + a32[0] - a32[2] - a32[3] - a32[4] - a32[5] + 3 * a32[7];
+
+	t[1] += t[0] >> 32; t[0] &= 0xffffffff;
+	t[2] += t[1] >> 32; t[1] &= 0xffffffff;
+	t[3] += t[2] >> 32; t[2] &= 0xffffffff;
+	t[4] += t[3] >> 32; t[3] &= 0xffffffff;
+	t[5] += t[4] >> 32; t[4] &= 0xffffffff;
+	t[6] += t[5] >> 32; t[5] &= 0xffffffff;
+	t[7] += t[6] >> 32; t[6] &= 0xffffffff;
+
+	o = t[7] >> 32; t[7] &= 0xffffffff;
+
+	t[0] += o;
+	t[3] -= o;
+	t[6] -= o;
+	t[7] += o;
+	t[1] += t[0] >> 32; t[0] &= 0xffffffff;
+	t[2] += t[1] >> 32; t[1] &= 0xffffffff;
+	t[3] += t[2] >> 32; t[2] &= 0xffffffff;
+	t[4] += t[3] >> 32; t[3] &= 0xffffffff;
+	t[5] += t[4] >> 32; t[4] &= 0xffffffff;
+	t[6] += t[5] >> 32; t[5] &= 0xffffffff;
+	t[7] += t[6] >> 32; t[6] &= 0xffffffff;
+
+	r[0] = (t[1] << 32) | t[0];
+	r[1] = (t[3] << 32) | t[2];
+	r[2] = (t[5] << 32) | t[4];
+	r[3] = (t[7] << 32) | t[6];
+}
+
+static void
+mont_basic(void)
+{
+	const unsigned long a[8] = {1, 0, 2, 3}, b[4] = {4, 5, 6, 7};
+	unsigned long am[8] = {}, bm[4] = {};
+	unsigned long r[8] = {5}; /* only 4 limbs are used, fill with garbage */
```

Comment:
This doesn't do what is stated: the `r[]` array is not filled with garbage, first element is filled with 5, others will be zeroed.